internal/pkg/nats/client.go

```diff
@@ -126,10 +139,13 @@ func (c *Client) Subscribe(topics []types.TopicChannel, messageErrors chan error
 		return fmt.Errorf("cannot subscribe with disconnected client")
 	}
 
+	c.subscriptionMutex.Lock()
+	defer c.subscriptionMutex.Unlock()
+
 	for _, tc := range topics {
 		s := TopicToSubject(tc.Topic)
 
-		_, err := c.connection.QueueSubscribe(s, c.config.QueueGroup, func(msg *nats.Msg) {
+		subscription, err := c.connection.QueueSubscribe(s, c.config.QueueGroup, func(msg *nats.Msg) {
```

Comment:
Not sure there's much real exposure here (maybe if a request ID gets reused but seems unlikely) but concurrent subscribe/unsubscribe to the same topic could be an issue.

Reply:
Nope, no real exposure since requestId (UUID) will keep it unique